backend/manager/modules/vdsbroker/src/main/java/org/ovirt/engine/core/vdsbroker/vdsbroker/VdsBrokerObjectsBuilder.java

```diff
@@ -1408,10 +1406,23 @@ public void updateVDSStatisticsData(VDS vds, Map<String, Object> struct) {
 
         vds.setBootTime(assignLongValue(struct, VdsProperties.bootTime));
 
+        // this method needs to be called after the free memory and hugepages are set to vds
+        updateMemoryUsage(vds);
         updateNumaStatisticsData(vds, struct);
         updateV2VJobs(vds, struct);
     }
 
+    private void updateMemoryUsage(VDS vds) {
+        if (vds.getPhysicalMemMb() == null || vds.getMemFree() == null || vds.getHugePages() == null) {
+            return;
+        }
+        int memTotal = vds.getPhysicalMemMb() - HugePageUtils.totalHugePageMemMb(vds);
+        long memFree = vds.getMemFree();
+        long memUsage = 100 - Math.round(100 * memFree / (double) memTotal);
+
+        vds.setUsageMemPercent((int) memUsage);
+    }
```

Comment:
I am starting to have doubts about this calculation... Basically, we are computing the CPU usage in the shared memory (non hugepages). But on the UI this results in hight CPU usage displayed even though nothing runs on the host. 
![image](https://user-images.githubusercontent.com/2078072/172155080-8efe4183-1f6d-4d63-8406-f9a1049d087b.png)

Maybe instead of excluding the hugepages from the total, we should include the free hugepages in the calculating the used memory like this:
```
    private void updateMemoryUsage(VDS vds) {
        if (vds.getPhysicalMemMb() == null || vds.getMemFree() == null || vds.getHugePages() == null) {
            return;
        }
        int memTotal = vds.getPhysicalMemMb();
        long memFree = vds.getMemFree() + HugePageUtils.totalHugePageFreeMemMb(vds);
        long memUsage = 100 - Math.round(100 * memFree / (double) memTotal);

        vds.setUsageMemPercent((int) memUsage);
    }
```

Reply:
> I am starting to have doubts about this calculation... Basically, we are computing the CPU usage in the shared memory (non hugepages). But on the UI this results in hight CPU usage displayed even though nothing runs on the host. ![image](https://user-images.githubusercontent.com/2078072/172155080-8efe4183-1f6d-4d63-8406-f9a1049d087b.png)

yeah, that's why we thought of having the warning only when there's a VM running on the host

> 
> Maybe instead of excluding the hugepages from the total, we should include the free hugepages in the calculating the used memory like this:
> 
> ```
>     private void updateMemoryUsage(VDS vds) {
>         if (vds.getPhysicalMemMb() == null || vds.getMemFree() == null || vds.getHugePages() == null) {
>             return;
>         }
>         int memTotal = vds.getPhysicalMemMb();
>         long memFree = vds.getMemFree() + HugePageUtils.totalHugePageFreeMemMb(vds);
>         long memUsage = 100 - Math.round(100 * memFree / (double) memTotal);
> 
>         vds.setUsageMemPercent((int) memUsage);
>     }
> ```

we have two kinds of "memory" and it's hard to treat them uniformly - but as Liran wrote, by aligning to how it's represented by the platform, it would be easier to justify

Reply:
Lets try to define what we want to achieve. There are 2 places where we deal with the memory usage:
- Audit log -> VDS_HIGH_MEM_USE - we want to warn the user that the host is running out of the memory.
- UI -> Host list -> Memory usage - we want to show the overall status of the memory usage on the host

If that is the case, I think we should include hugepages in the free memory (I don't think we should be bound by how platform does it, this is our business logic).

BTW, this is quite similar to what we talked about how the CPU usage should be shown to the user on the Host list page. We decided to show the result for all CPUs (shared + exclusively pinned) because at Host list page we want to see the overall status of the host. This could apply to the memory so that we should show the memory graph for both, normal memory + hugepages.

Reply:
yeah, I was referring only to the warning - not the graph
it sounds ok to include the hugepages in the calculation for the graph and not to include it for the warnings